### PR TITLE
Feature/hold payment modal

### PR DESCRIPTION
**Pull request by mugnivenko**
https://app.clickup.com/t/1k71gxj

### CODE REVIEW

File: api/PayRun/index.js [hunk 1]
```diff
@@ -7,4 +7,10 @@ const PAYRUNS_URL = `${BASE_URL}/v1/payruns`;
 const sendPayRunsRequest = (url = '', data, method = requestMethods.post) =>
   axiosRequest({ url: `${PAYRUNS_URL}${url ? `/${url}` : ''}`, data, method });
 
-export const createDraftPayRun = ({ type, paidUpToDate }) => sendPayRunsRequest('', { type, paidUpToDate });
\ No newline at end of file
+const sendInvoiceRequest = ({ payRunId, invoiceId, params }) =>
+  sendPayRunsRequest(`${payRunId}/invoices/${invoiceId}/hold`, params);
+
+export const createDraftPayRun = ({ type, paidUpToDate }) => sendPayRunsRequest('', { type, paidUpToDate });
```
Comment (akeamwoodsND) [inline api/PayRun/index.js:13]: don't need '' in sendPayRunsRequest

File: components/HackneyDS/Dialog/index.js [hunk 2]
```diff
@@ -30,12 +30,13 @@ export default function Dialog({
     };
   }, [onClose]);
 
+  if (!isOpen) return null;
+
   if (!windowState) return null;
```
Comment (akeamwoodsND) [inline components/HackneyDS/Dialog/index.js:35]: if(!isOpen | !windowState)

File: components/HackneyDS/Select/index.js [hunk 3]
```diff
@@ -26,7 +26,6 @@ const Select = (props) => {
     <div className="select-container">
       <select
         style={style}
-        disabled={isDisabled}
```
Comment (akeamwoodsND) [inline components/HackneyDS/Select/index.js:29]: why have you removed this?

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]

File: components/HackneyDS/Select/index.js (hunk 3 above)
Comment (mugnivenko) [reply components/HackneyDS/Select/index.js:29]: because we have disabled prop

Comment (mugnivenko) [reply components/HackneyDS/Select/index.js:29]: image

Comment (akeamwoodsND) [reply components/HackneyDS/Select/index.js:29]: but you're not passing 'disabled' into the select component?

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
Screenshot 2021-11-12 at 10 52 37
Modal needs to be centred

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
looking good mostly though

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
no validation? need error message to show errors
Screenshot 2021-11-13 at 17 34 31

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
That gap shouldn't be there 
Screenshot 2021-11-15 at 12 41 48

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
Screenshot 2021-11-15 at 12 47 30
change validation text to 'Please select an option' and for text area 'Please enter a reason'

File: pages/payruns/[guid]/index.js [hunk 4]
```diff
@@ -67,6 +70,13 @@ const SinglePayRun = () => {
 
   return (
     <Container>
+      <HoldPayment
```
Comment (akeamwoodsND) [inline pages/payruns/[guid]/index.js:73]: move this component to inside PayRunItem, pass update through to PayRunItem as a prop and move the [invoidId,setInvoiceId] line 38 to inside payRunItem

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]

**Pull request merged**